### PR TITLE
fixing misspellings, fixing IneffAssign reported issues

### PR DESCRIPTION
**Pull request by mikegray**
Continuing to fix some of the issues discovered through the GoReportCard. 

- 2 misspellings
- all of the in-effectual assignments
- also removing encode-perf-rig since codec-perf-rig serves this purpose better

### CODE REVIEW

**Comment by mikegray** [comment]
@cmasone-attic PTAL if you don't mind

File: go/perf/codec-perf-rig/main.go [hunk 1]
```diff
@@ -168,9 +168,8 @@ func buildListIncrementally(count uint64, createFn createValueFn) types.Collecti
 }
 
 func readList(c types.Collection) {
-	var outValue types.Value
 	c.(types.List).IterAll(func(v types.Value, idx uint64) {
-		outValue = v
+		_ = v
```
Comment (cmasone-attic) [inline go/perf/codec-perf-rig/main.go]: Can you provide a comment in these three functions explaining why the bodies aren't just empty?

**Comment by cmasone-attic** [comment]
LGTM

**Comment by aboodman** [comment]
I think some of those ineffectual assigns in perf rig were required to get realistic numbers out of perf rig.

**Comment by cmasone-attic** [comment]
I woudn't be surprised if that were so, to keep the compiler from
optimizing stuff out. If so, then we should leave a comment indicating that

**Comment by mikegray** [comment]
I am not seeing any noticeable performance difference. I thought the same thing originally - "oh, those assignments (while not being used) are to ensure the code doesn't get optimized away".  If the compiler is smart, it would already be seeing the same thing as IneffAssign and optimizing these calls away even without the code changes. I don't think it can optimize away the IterAll calls which are really all that matters for the perf results. The cost of assigning the variables or calling the callback functions is tiny. I decided to run it with a larger count to still try to tease out any change of performance. The only numbers that should change are in the scan columns - the readList/readSet/readMap calls.  Do you see anything below?

Results before this change (from origin/master):
```
$ ./codec-perf-rig --count 1000000
Testing List: 		build 1000000			scan 1000000			insert 20000
numbers (8 B)		805 ms (9.93 MB/s)		366 ms (21.82 MB/s)		26151 ms (0.01 MB/s)
strings (32 B)		1858 ms (17.21 MB/s)		323 ms (98.88 MB/s)		20093 ms (0.03 MB/s)
structs (64 B)		4153 ms (15.41 MB/s)		1186 ms (53.93 MB/s)		9383 ms (0.14 MB/s)

Testing Set: 		build 1000000			scan 1000000			insert 20000
numbers (8 B)		880 ms (9.09 MB/s)		418 ms (19.11 MB/s)		27484 ms (0.01 MB/s)
strings (32 B)		2178 ms (14.69 MB/s)		307 ms (104.17 MB/s)		6995 ms (0.09 MB/s)
structs (64 B)		18309 ms (3.50 MB/s)		1138 ms (56.23 MB/s)		12216 ms (0.10 MB/s)

Testing Map: 		build 1000000			scan 1000000			insert 20000
numbers (8 B)		1839 ms (4.35 MB/s)		651 ms (12.27 MB/s)		108207 ms (0.00 MB/s)
strings (32 B)		4264 ms (7.50 MB/s)		566 ms (56.46 MB/s)		20755 ms (0.03 MB/s)
structs (64 B)		24367 ms (2.63 MB/s)		2362 ms (27.09 MB/s)		18560 ms (0.07 MB/s)

Testing Blob: 		build 33 MB			scan 33 MB
			802 ms (41.80 MB/s)		423 ms (79.21 MB/s)
```

Results after this change (from this branch - created directly from origin/master this morning) - should be faster now if the compiler wasn't already optimizing away the ineffassign results:
```
$ ./codec-perf-rig --count 1000000
Testing List: 		build 1000000			scan 1000000			insert 20000
numbers (8 B)		832 ms (9.61 MB/s)		408 ms (19.57 MB/s)		25680 ms (0.01 MB/s)
strings (32 B)		2006 ms (15.95 MB/s)		393 ms (81.35 MB/s)		20983 ms (0.03 MB/s)
structs (64 B)		4712 ms (13.58 MB/s)		1299 ms (49.26 MB/s)		10085 ms (0.13 MB/s)

Testing Set: 		build 1000000			scan 1000000			insert 20000
numbers (8 B)		794 ms (10.06 MB/s)		387 ms (20.65 MB/s)		32009 ms (0.00 MB/s)
strings (32 B)		2359 ms (13.56 MB/s)		328 ms (97.49 MB/s)		7839 ms (0.08 MB/s)
structs (64 B)		20817 ms (3.07 MB/s)		1180 ms (54.23 MB/s)		13735 ms (0.09 MB/s)

Testing Map: 		build 1000000			scan 1000000			insert 20000
numbers (8 B)		1963 ms (4.07 MB/s)		673 ms (11.88 MB/s)		112845 ms (0.00 MB/s)
strings (32 B)		4351 ms (7.35 MB/s)		593 ms (53.96 MB/s)		21989 ms (0.03 MB/s)
structs (64 B)		25662 ms (2.49 MB/s)		2431 ms (26.32 MB/s)		18100 ms (0.07 MB/s)

Testing Blob: 		build 33 MB			scan 33 MB
			798 ms (42.01 MB/s)		427 ms (78.46 MB/s)
```

File: go/perf/codec-perf-rig/main.go (hunk 1 above)
Comment (mikegray) [inline go/perf/codec-perf-rig/main.go]: The results in the tables above or below that I just included were with the function bodies completely empty. I will submit that as a patch here so then we don't even have to comment on why they are not empty.

**Comment by mikegray** [comment]
@cmasone-attic, @aboodman PTAL and let me know what you think. I can put the assignments back if you are still concerned after reviewing the timing information above.

**Comment by cmasone-attic** [comment]
...huh. I don't see any difference. @aboodman ?

**Comment by aboodman** [comment]
I just vaguely remember this. If the data doesn't show any difference I'm
fine removing them.

**Pull request merged**